rpcserver.go

```diff
@@ -3107,10 +3137,93 @@ func (r *rpcServer) QueryAssetStats(ctx context.Context,
 		return nil, err
 	}
 
-	snapshots := assetStats.SyncStats
 	resp := &unirpc.UniverseAssetStats{
-		AssetStats: fn.Map(snapshots, marshalAssetSyncSnapshot),
+		AssetStats: make(
+			[]*unirpc.AssetStatsSnapshot, len(assetStats.SyncStats),
+		),
+	}
+	for idx, snapshot := range assetStats.SyncStats {
+		resp.AssetStats[idx] = marshalAssetSyncSnapshot(snapshot)
+		resp.AssetStats[idx].GenesisTimestamp = r.getBlockTimestamp(
+			ctx, snapshot.GenesisHeight,
+		)
 	}
 
 	return resp, nil
 }
+
+// getBlockTimestamp returns the timestamp of the block at the given height.
+func (r *rpcServer) getBlockTimestamp(ctx context.Context,
+	height uint32) int64 {
+
+	// Shortcut any lookup in case we don't have a valid height in the first
+	// place.
+	if height == 0 {
+		return 0
+	}
+
+	cacheTS, err := r.blockTimestampCache.Get(height)
+	if err == nil {
+		return int64(cacheTS)
+	}
+
+	hash, err := r.cfg.Lnd.ChainKit.GetBlockHash(ctx, int64(height))
+	if err != nil {
+		return 0
+	}
+
+	block, err := r.cfg.Lnd.ChainKit.GetBlock(ctx, hash)
```

Comment:
Non-blocking: I saw that this wasn't exposed via `ChainKit` but it seems like here (and elsewhere), we would want to use something like `GetBlockHeader()` and not fetch a whole block?

Or is that not exposed because some chain backends don't support that?

Reply:
We do have a `GetBlockHeader()` method for every chain backend. The reason we can't use it here is because there currently is no RPC in `lnd` that exposes the call.
I agree that we'll want to add this as an optimization. I'll see if I can get a PR in for 0.17.